src/compile/data/parse.ts

```diff
@@ -160,8 +191,15 @@ Formula From Sort Array
 export function parseData(model: Model): DataComponent {
   let head = parseRoot(model, model.component.data.sources);
 
-  const outputNodes = model.component.data.outputNodes;
-  const outputNodeRefCounts = model.component.data.outputNodeRefCounts;
+  const {outputNodes, outputNodeRefCounts} = model.component.data;
+  const ancestorParse = model.parent ? model.parent.component.data.ancestorParse.clone() : new AncestorParse();
```

Comment:
I don't understand why do you have to clone here?

Reply:
I don't have to. I guess we could just reuse the same object and modify it as we go down the tree.

Reply:
I was wrong. We have to as explained earlier.

Reply:
> We have to as explained earlier.

Can you explain again?  I don't know where you explain it as this PR is pretty complicated.

The rest of the "parse" codebase produce something as as part of component. 

Isn't it weird that this `ancestorParse` seems to be just floating within the parse codebase?

Reply:
At very least, this line deserves a better comment why this is being cloned.